src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -258,30 +258,39 @@ void BindType(NpgsqlTypeMapping mapping, NpgsqlConnector connector, bool throwOn
             if (!found)
             {
                 var msg = $"A PostgreSQL type with the name {mapping.PgTypeName} was not found in the database";
-                if (throwOnError)
+                if (externalCall)
                     throw new ArgumentException(msg);
                 Log.Debug(msg);
                 return;
             }
             else if (pgType == null)
             {
                 var msg = $"More than one PostgreSQL type was found with the name {mapping.PgTypeName}, please specify a full name including schema";
-                if (throwOnError)
+                if (externalCall)
                     throw new ArgumentException(msg);
                 Log.Debug(msg);
                 return;
             }
             else if (pgType is PostgresDomainType)
             {
                 var msg = "Cannot add a mapping to a PostgreSQL domain type";
-                if (throwOnError)
+                if (externalCall)
                     throw new NotSupportedException(msg);
                 Log.Debug(msg);
                 return;
             }
 
             var handler = mapping.TypeHandlerFactory.Create(pgType, connector.Connection);
             BindType(handler, pgType, mapping.NpgsqlDbType, mapping.DbTypes, mapping.ClrTypes);
+
+            if (externalCall)
+                foreach (var domain in DatabaseInfo.DomainTypes)
+                    if (domain.BaseType.OID == pgType.OID)
```

Comment:
nit: Could lift this into `.Where(d => d.BaseType.OID == pgType.OID)`.

Reply:
It will decrease performance because of delegate call and it will allocate memory for a closure.

Reply:
Oh, those are fair points. Though, how hot is this path in practice?

Reply:
It will also allocate a `Where` iterator each time you reach the loop. And this happens when you map a composite or an enum type on an opened connection.